### PR TITLE
Feature/jupyter

### PR DESCRIPTION
**Pull request by vanpelt**
Yo dude, would love to have your eyes on this.  Finally got stuff working in Jupyter.

### CODE REVIEW

File: wandb/jupyter.py [hunk 1]
```diff
@@ -1,11 +1,77 @@
...
+    """A class that only logs metrics after `wandb.log` has been called and stops logging at cell completion"""
+
+    def __init__(self):
+        self.paused = True
+
+    def start(self):
+        if self.paused:
+            self.api = InternalApi()
+            # TODO: there's an edge case where shutdown isn't called
+            self.api._file_stream_api = None
```
Comment (adrianbg) [inline wandb/jupyter.py]: Why is this necessary?

File: wandb/run_manager.py [hunk 2]
```diff
@@ -519,6 +520,10 @@ def mirror_stdout_stderr(self):
         io_wrap.SimpleTee(sys.stderr, streaming_log.TextStreamPusher(
             fs_api, OUTPUT_FNAME, prepend_timestamp=True, line_prepend='ERROR'))
 
+    def unmirror_stdout_stderr(self):
+        sys.stdout.write = sys.stdout.orig_write
+        sys.stderr.write = sys.stderr.orig_write
```
Comment (adrianbg) [inline wandb/run_manager.py]: Might be better to put this stuff in the SimpleTee class

File: wandb/jupyter.py [hunk 3]
```diff
@@ -1,11 +1,77 @@
...
         self.run = run or wandb.run
 
     def _repr_html_(self):
-        url = self.run.get_url()+"?jupyter=true"
+        state = "running"
+        if self.run._jupyter_agent == None:
+            state = "no_agent"
+        elif self.run._jupyter_agent.paused:
+            state = "paused"
+        url = self.run.get_url() + "?jupyter=true&state=" + state
```
[the remark this answers is not in the document]
Comment (vanpelt) [reply wandb/jupyter.py]: Since we’re only putting in known strings without illegal chars and get_url returns only the path, we’re good

File: wandb/jupyter.py (hunk 1 above)
Comment (vanpelt) [reply wandb/jupyter.py]: I believe there’s a case where shutdown is never called.  Should probably just store the state in a global and call shutdown again in this case...

File: wandb/__init__.py [hunk 4]
```diff
@@ -287,9 +301,9 @@ def save(path):
     return os.symlink(os.path.abspath(path), os.path.join(run.dir, file_name))
 
 
-def monitor(display=True, options={}):
-    """Starts syncing with W&B if you're in Jupyter.  Displays your W&B charts live in a Jupyter notebook
-    by default.  Display can be disabled by passing display=False.
+def monitor(options={}):
+    """Starts syncing with W&B if you're in Jupyter.  Displays your W&B charts live in a Jupyter notebook.
+    It's currently a context manager for legacy reasons.
```
[the remark this answers is not in the document]
Comment (vanpelt) [reply wandb/__init__.py]: Old docs reference the context manager so we’ll keep it around until a major release.

File: wandb/run_manager.py (hunk 2 above)
Comment (vanpelt) [reply wandb/run_manager.py]: Agreed I’ll do it in a separate pull.
